ibis/backends/flink/ddl.py

```diff
@@ -0,0 +1,154 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import sqlglot as sg
+
+from ibis.backends.base.sql.ddl import (
+    CreateTableWithSchema,
+    DropObject,
+    _CreateDDL,
+    _format_properties,
+    _is_quoted,
+    is_fully_qualified,
+)
+from ibis.backends.base.sql.registry import quote_identifier
+
+if TYPE_CHECKING:
+    import ibis.expr.schema as sch
+
+
+class _CatalogAwareBaseQualifiedSQLStatement:
+    def _get_scoped_name(
+        self, obj_name: str, database: str | None = None, catalog: str | None = None
+    ) -> str:
+        if is_fully_qualified(obj_name):
+            return obj_name
+        if _is_quoted(obj_name):
+            obj_name = obj_name[1:-1]
+        return sg.table(obj_name, db=database, catalog=catalog, quoted=True).sql(
+            dialect="hive"
+        )
+
+
+class CreateTableFromConnector(
+    _CatalogAwareBaseQualifiedSQLStatement, CreateTableWithSchema
+):
+    def __init__(
+        self,
+        table_name: str,
+        schema: sch.Schema,
+        tbl_properties: dict,
+        database: str | None = None,
+        catalog: str | None = None,
+        temp: bool = False,
+        **kwargs,
+    ):
+        super().__init__(
+            table_name=table_name,
+            database=database,
+            schema=schema,
+            table_format=None,
+            format=None,
+            path=None,
+            tbl_properties=tbl_properties,
+            **kwargs,
+        )
+        self.catalog = catalog
+        self.temp = temp
+
+    def _storage(self) -> str:
+        return f"STORED AS {self.format}" if self.format else None
+
+    def _format_tbl_properties(self) -> str:
+        return f"WITH {_format_properties(self.tbl_properties)}"
+
+    @property
+    def _prefix(self) -> str:
+        if self.temp is True:
+            # This is not documented in Flink's documentation
+            return "CREATE TEMPORARY TABLE"
+        else:
+            return "CREATE TABLE"
```

Comment:
```suggestion
        return f"CREATE {"TEMPORARY " if self.temp else ""} TABLE"
```
Nit: I like the condensed expression :)

Reply:
I actually find the current way more readable, but either works. Personally I'd go with:

```python
modifier = " TEMPORARY" if self.temp else ""
return f"CREATE{modifier} TABLE"
```